drivers/modem/gsm_ppp.c

```diff
@@ -913,13 +909,18 @@ static void mux_setup(struct k_work *work)
 
 		gsm_finalize_connection(gsm);
```

Comment:
Need to handle gsm->state set in `gsm_finalize_connection()` and fails, easier if change `gsm_finalize_connection` into a work.